examples/async_tasks/async_compute.rs

```diff
@@ -76,12 +78,12 @@ fn spawn_tasks(mut commands: Commands, thread_pool: Res<AsyncComputeTaskPool>) {
 /// removes the task component from the entity.
 fn handle_tasks(
     mut commands: Commands,
-    mut transform_tasks: Query<(Entity, &mut Task<Transform>)>,
+    mut transform_tasks: Query<(Entity, &ComputeTransformTask)>,
     box_mesh_handle: Res<BoxMeshHandle>,
     box_material_handle: Res<BoxMaterialHandle>,
 ) {
-    for (entity, mut task) in transform_tasks.iter_mut() {
-        if let Some(transform) = future::block_on(future::poll_once(&mut *task)) {
```

Comment:
Does it make sense to still keep this as an example how to handle `Task`s emerging from `TaskPool::spawn`?

Reply:
Are there any patterns that directly using tasks enables over using PollableTask? My understanding of the point of this pr is that users should be avoiding the use of block_on in their code.

Reply:
Then my next question would be why keep `Task` in the public API? I'd say 
- either remove it and make `TaskPool::spawn` `pub(crate)` or something
- or keep an example how to use it?

Reply:
We need the current version of Task if we want to detach and return multiple values from a task. We might want an example of that.

Reply:
What do you mean by "return multiple values"? How would I do that? If I understood well enough what you mean, I could still add such an example.

Reply:
if you use a channel to send values between the task and a system, you can use that channel to send a values to the system multiple times

Reply:
So that would basically be the same as the [external_source_external_thread example](https://github.com/bevyengine/bevy/blob/1a85fb5ea3e30bf5ffca6a82c06c0c73b37bd93e/examples/async_tasks/external_source_external_thread.rs)? 

I thought a task is something that works for a while in the background and then carries a single result. What you propose would then be a `Task<()>` or do I not quite get what you mean?

Reply:
oh, I wasn't aware that example existed. I'm not sure people should be spawning threads outside the threadpool, since it'll lead to more contention. But I'm also not sure they shouldn't.

It might be best to kick the can down the road and leave task public for now. Maybe make an issue to make a decision if it should be make private or make an example for how to use Task?

Reply:
@alice-i-cecile any opinion?

Reply:
-> #4198